src/lib/graphql_basic_scalars/graphql_basic_scalars.ml

```diff
@@ -19,6 +25,62 @@ module type Json_intf = sig
   val typ : unit -> ('a, t option) Graphql_async.Schema.typ
 end
 
+module type Test_Intf = sig
+  type t
+
+  val gen : t Base_quickcheck.Generator.t
+
+  val sexp_of_t : t -> Ppx_sexp_conv_lib.Sexp.t
+
+  val compare : t -> t -> int
+end
+
+let json_from_response = function
+  | Ok (`Response data) ->
+      Async_kernel.return data
+  | Ok (`Stream stream) ->
+      Async_kernel.Pipe.to_list stream
+      >>| fun lst ->
+      `List
+        Core_kernel.(List.map lst ~f:(fun x -> Option.value_exn (Result.ok x)))
+  | Error err ->
+      Async_kernel.return err
+
+let test_query schema ctx query (f_test : Yojson.Basic.t -> unit) : unit =
+  Thread_safe.block_on_async_exn (fun () ->
```

Comment:
Done in [c46ed9d](https://github.com/MinaProtocol/mina/pull/11667/commits/c46ed9d4edfdb8ec9f7f9fbb1cb7d9466e725635).
The various graphql scalars are now defined in functors so that they can be instantiated using multiple GraphQL `Schema` modules. This way the tests can use a pure one while the rest of the application uses the async one.